### PR TITLE
replace memset with value-initialization

### PR DESCRIPTION
**Pull request by dagitses**
Stack created with [Sapling](https://sapling-scm.com). Best reviewed with [ReviewStack](https://reviewstack.dev/pytorch/pytorch/pull/90048).
* #89865
* #89852
* #89851
* __->__ #90048

replace memset with value-initialization

Summary:
This is equivalent to zero initialization for any members that are
scalar or have implicit default constructors.

Note that aside from the reset at the beginning, blockmask and
philox_args are not touched by this function.

Test Plan: Rely on CI.

### CODE REVIEW

**Review by drisspg — APPROVED** [review]

**Review by malfet — APPROVED** [review]

File: aten/src/ATen/native/transformers/cuda/flash_attn/fmha_api.cpp [hunk 1]
```diff
@@ -64,7 +64,7 @@ void set_params_fprop(FMHA_fprop_params &params,
                       bool is_causal) {
 
     // Reset the parameters
-    memset(&params, 0, sizeof(params));
+    params = {};
```
Comment (malfet) [inline aten/src/ATen/native/transformers/cuda/flash_attn/fmha_api.cpp:67]: Hmm, technically those are not equivalent, are they?  I.e. default constructor might not set the value for all parameters or might set it to values other than 0?

Comment (dagitses) [reply aten/src/ATen/native/transformers/cuda/flash_attn/fmha_api.cpp:67]: Correct, these are not directly equivalent. memset is not allowed here though.

The rules for this are somewhat complicated, but most fields should be zeroed out with this. Nevertheless, most fields are written directly in this function anyways, with the exception of those that are called out in the summary above.

**Pull request closed without merging**